### PR TITLE
Upgrade publisher to React 18 to match agency-dashboard + move eslint config to parent directory

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

See: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

- Due to issues caused by mismatching React versions, upgrade publisher to React 18 to match with agency-dashboard
  - I changed ReactDOM.render to the new `createRoot` method. See upgrade guide.
  - `children` is no longer a prop of `React.ReactNode`. Either explicitly define `children` in props or use `React.PropsWithChildren` instead.
  - I moved React.StrictMode to the child of the `AuthWall` component, since in the new Strict Mode, React will `automatically unmount and remount every component, whenever a component mounts for the first time, restoring the previous state on the second mount` !!! This caused an issue where Auth0.authenticate gets called twice in rapid succession, which triggers a 403 error.
- move eslint config to parent directory so it can be shared with all projects that run `yarn eslint`.
- Configure eslint for `common` and `agency-dashboard`, using the new shared eslint config.
- Remove some unneeded files in `agency-dashboard`.
- Remove example `SharedComponent` from `common`.
- Upgraded jest tests + added linting agency-dashboard and common in CI

I was running into issues of mismatching React versions, so I tried upgrading `publisher` to React 18 to see if it was easy to do. It ended up requiring a bit of work to upgrade but it only took an hour for me to get to this point and it seems like the upgrade is complete now? If we run into more issues that are not trivial, I can shelve this and just downgrade our agency-dashboard to React 17.

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by lilidworkin** [comment]
Oh wow, thanks for taking this on, @terryttsai ! I'll take a closer look at this later, but generally am not at all opposed to this. We should just test it rigorously before deploying, especially to catch other issues relating to mounting/unmounting.

File: publisher/src/index.tsx [hunk 1]
```diff
@@ -109,16 +109,27 @@ const GlobalStyle = createGlobalStyle`
...
+const rootContainer = document.getElementById("root");
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const root = createRoot(rootContainer!); // "!"" is required for use with TypeScript
+
+root.render(
+  <StoreProvider>
+    <GlobalStyle />
+    <AuthWall>
+      {/** Moved StrictMode to not mount, unmount, and remount AuthWall on the first render
```
Comment (lilidworkin) [inline publisher/src/index.tsx:121]: Nice comment here!

**Comment by lilidworkin** [comment]
I think @mxosman is probably the right person to do the more thorough review here. @terryttsai do you think we need to sanity check all functionality of the app first before merging? How likely do you think it is that another issue like the Auth0 authenticate one cropped up somewhere?

**Comment by terryttsai** [comment]
We don't have any more warnings / errors but I would recommend sanity checking the functionality of the app before merging, I'll do that! 

Right now Auth0 is working fine but we should test its functionality like logout, if no issue pops up in testing I would rule out Auth0 issues.

**Comment by mxosman** [comment]
Yeah, I'd like to take some time with this and read up on the new React v18 changes before I would feel comfortable with the upgrade. Will give it a read through and do a sanity check as well. Thank you for the descriptive PR!

**Comment by terryttsai** [comment]
I'm curious if you think the change to [automatic batching](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#automatic-batching) will cause any issues for us

**Comment by terryttsai** [comment]
TODO: update tests to handle warning: "Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot"

**Comment by mxosman** [comment]
> I'm curious if you think the change to [automatic batching](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#automatic-batching) will cause any issues for us

It might - hard to tell without testing. Nevertheless, it's an awesome performance improvement.

**Review by mxosman — APPROVED** [review]
Terry - thank you for going through all of the set up you went through and for syncing to discuss the upgrade! Did a manual test and no issues popped up - I have no concerns and look forward to all of the performance boosts we get for free! 
🚀

**Comment by terryttsai** [comment]
Good syncing w you and thanks!

**Pull request merged**